### PR TITLE
chore: Merge existing migrations - Methods & Properties

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes https://jira.tools.sap/browse/CXSPA-712

### CODE REVIEW

File: docs/migration/5_0.md [hunk 1]
```diff
@@ -407,8 +407,6 @@ The following keys have been added to `configurator-common.ts`:
 
 ### AuthHttpHeaderService
 
-- The `refreshInProgress` property was removed. Use `refreshInProgress$` Observable instead.
-- The `handleExpiredToken` method was removed. Use `getValidToken` instead.
 - In `handleExpiredAccessToken` method third parameter `initialToken` is now required.
```
Comment (bgambocjaviniar) [inline docs/migration/5_0.md:410]: Reminder:

Ask Patrick about optional params. I recall these were supported before, but currently it does not handle the case where it was optional and now required

**Review by bgambocjaviniar — COMMENTED** [review]
TODO:

- ~~remind ask Patrick to push the old and new changes in epic, so that I can pull the json for `SavedCartEventBuilder` instead of writing it manually~~ (nvm I'm blind)
- handling optional param that became required. For example, handleExpiredAccessToken in AuthHttpHeaderService

File: docs/migration/5_0.md [hunk 2]
```diff
@@ -507,12 +507,6 @@ The following keys have been added to `configurator-common.ts`:
 - `resultsFound` key has been removed.
 - Following keys `storesFound`, `ariaLabelCountriesCount` have been added for screen reader improvements.
 
-### BadRequestHandler
-
-- Moved `handleVoucherOperationError` method from `core` to the new handler `BadVoucherRequestHandler` in cart-lib
-
-- Method `handleBadLoginResponse` has been modified to determine error message for email update or password change based on url pathname.
```
Comment (bgambocjaviniar) [inline docs/migration/5_0.md:558]: I don't think this is needed to be said

**Comment by bgambocjaviniar** [comment]
> TODO:
> 
> - ~~remind ask Patrick to push the old and new changes in epic, so that I can pull the json for `SavedCartEventBuilder` instead of writing it manually~~ (nvm I'm blind)
> - handling optional param that became required. For example, handleExpiredAccessToken in AuthHttpHeaderService

This will be handled in a separate ticket

**Review by plabadie — APPROVED** [review]

**Pull request merged**